src/graph/graphcontext.c

```diff
@@ -417,18 +424,45 @@ void GraphContext_RemoveAttribute(GraphContext *gc, Attribute_ID id) {
 bool GraphContext_HasIndices(GraphContext *gc) {
 	ASSERT(gc != NULL);
 
-	uint schema_count = array_len(gc->node_schemas);
-	for(uint i = 0; i < schema_count; i++) {
-		if(Schema_HasIndices(gc->node_schemas[i])) return true;
-	}
+	const bool has_node_indices = GraphContext_NodeIndexCount(gc);
+	const bool has_edge_indices = GraphContext_EdgeIndexCount(gc);
+
+	return has_node_indices || has_edge_indices;
+}
+
+static uint64_t _count_indices_from_schemas(const Schema** schemas) {
+	ASSERT(schemas);
+	uint64_t count = 0;
 
-	schema_count = array_len(gc->relation_schemas);
-	for(uint i = 0; i < schema_count; i++) {
-		if(Schema_HasIndices(gc->relation_schemas[i])) return true;
+	const uint32_t length = array_len(schemas);
+	for (uint32_t i = 0; i < length; ++i) {
+		const Schema *schema = schemas[i];
+		ASSERT(schema);
+		if (!schema) {
```

Comment:
Please remove this excessive if condition

Reply:
It is excessive because of the DEBUG ASSERT just above it.
We test our code in DEBUG mode, making sure this path is covered, after which in production (RELEASE mode) we don't want to take on the overhead of performing this additional check